source/Halibut/Util/SilentStreamDisposer.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Halibut.Transport.Streams;
+
+namespace Halibut.Util
+{
+    public class SilentStreamDisposer : IAsyncDisposable
+    {
+        readonly Stream streamToDispose;
+        readonly Action<Exception> onFailure;
+
+        public SilentStreamDisposer(Stream streamToDispose, Action<Exception> onFailure = null)
+        {
+            this.streamToDispose = streamToDispose;
+            this.onFailure = onFailure;
+        }
+        
+        public async ValueTask DisposeAsync()
+        {
+            try
+            {
+                await streamToDispose.DisposeAsync();
+            }
+            catch (Exception ex)
+            {
+                onFailure?.Invoke(ex);
```

Comment:
Not sure how valuable the exception here is, but if we want to log something here, then we could do this at the Verbose level.